enhancements/authentication/configuring-webhook-token-authenticators.md

```diff
@@ -0,0 +1,159 @@
+be used for the purposes of configuring webhook token authenticators, where we would just expect references
+to secrets, that would have to be checked.
+
+The configuration mentioned above leaves quite a lot of freedom in hands of users as the referenced secrets
+would be quite general kubeconfigs which makes users be able to configure fields that could make the
+kube-apiserver access user-specified paths, which may not be desirable (although could be ruled out
+by validation).
+
+An alternative is to handle the configuration by introducing a new cluster-wide CRD:
+```
+apiVersion: config.openshift.io/v1
+kind: TokenAuthenticator
+spec:
+  ca: CM reference
+  server: string (valid https:// URL)
+  clientCertificate: CM reference
+  clientKey: Secret reference
+```
+
+#### Backend
+
+__cluster-authentication-operator__
+- observes the changes in the `authentication/cluster` resource (or observes new `TokenAuthenticator` object)
+  and performs validation of any references
+- it pushes any webhook authenticator that passes the validation to `openshift-config-managed/webhooktokenauthenticators`
+  secret, where `key=authenticator_name` where `authenticator_name` is either the name of the referenced secret or
+  the CR
+- if no webhook authenticators pass validation, the operator keeps the previous state
+
+__kube-apiserver-operator__
+- observes the `openshift-config-managed/webhooktokenauthenticators` secret, on a change it syncs it to the
+  openshift-kube-apiserver namespace and redeploys the kube-apiserver static pods while specifying the new
+  mounted files in the `webhookTokenAuthenticators` [config field](https://github.com/openshift/api/blob/992bc3a41fe6c4e5bcf27dc8080f9e5c114c33fc/kubecontrolplane/v1/types.go#L146).
+
+### Risks and Mitigations
+
+There are a couple of issues that root in the very fact of making it possible of
+replacing the oauth-server with a custom token authenticator:
+
+- `oc` and `oauth-proxy` rely on the _/.well-known/oauth-authorization-server_ endpoint
+  and if the integrated oauth-server is turned off and the `oauthMetadata` is not set
+  properly, things like grafana and prometheus won't be accessible
+- function of selfubjectaccessreviews and the "users/~" echo endpoint may not work
+  properly
+- ServiceAccounts may not be usable as OAuth clients anymore
+
+## Design Details
+
+### Test Plan
+
+Since we will be making the cluster-authentication-operator use the `webhookTokenAuthenticatros`
+(in combination with `oauthMetadata`, which already works) to configure authentication using
+the integrated OAuth server, we will just re-use all the tests that we have already.
+
+### Upgrade / Downgrade Strategy
+
+A handover between the older and newer configuration needs to exist so that
+authentication does not get disrupted or broken during upgrade.
+
+A 4.(n-1) cluster-authentication-operator needs to add the code for synchronizing
```

Comment:
how about you have the authentication operator wait until the kas-o indicates it reached 4.4 before making a change.  Then on downgrade (which can be bumpy), the kas-o will update and ignore your entries and directly use the oauth server

Reply:
I think you're right and it makes sense to wait, I added that to the section, please see if it needs more changes